### PR TITLE
Fixed parsing of content-length in HTTP header if tab/white spaces.

### PR DESCRIPTION
**Pull request by thomasegense**
Added unittest and a new WARC-file to test resources.

### CODE REVIEW

File: src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java [hunk 1]
```diff
@@ -267,7 +265,7 @@ else if (headerLine.toLowerCase().startsWith("location:")) {
     }
     else if (headerLine.toLowerCase().startsWith("transfer-encoding:")) {                                      
       String transferEncoding=headerLine.substring(18).trim();
```
Comment (tokee) [inline src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java:267]: Why use `substring` here when `split` is used on line 255? Consistent solutions to the same problem makes the code easier to real. Also, the use of constants such as `18` makes it harder to determine if the code does what it is supposed to.

File: src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java [hunk 2]
```diff
@@ -254,7 +252,7 @@ private static void populateWarcSecondHeader(ArcEntry warcEntry, String headerLi
 
     }  //Content-Length: 31131
     else if (headerLine.toLowerCase().startsWith("content-length:")) {
-      String[] contentLine = headerLine.split(" ");
+      String[] contentLine = headerLine.split(":");
       long totalSize = Long.parseLong(contentLine[1].trim());
```
Comment (tokee) [inline src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java:256]: Under the assumption that the `content-length` follows [RFC 2616](https://www.rfc-editor.org/rfc/rfc2616#page-31), this should work.

File: src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java [hunk 3]
```diff
@@ -267,7 +265,7 @@ else if (headerLine.toLowerCase().startsWith("location:")) {
     }
     else if (headerLine.toLowerCase().startsWith("transfer-encoding:")) {                                      
       String transferEncoding=headerLine.substring(18).trim();
-      log.debug("transfer-encoding:"+transferEncoding);
+      //log.debug("transfer-encoding:"+transferEncoding);
       if (transferEncoding.equalsIgnoreCase("chunked")) {
```
Comment (tokee) [inline src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java:269]: This should be a contains as the transfer-encoding can have more than one value: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding

**Review by tokee — COMMENTED** [review]
Up to you if you want to change based on this review. Functionally it works.

File: src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java (hunk 1 above)
Comment (thomasegense) [reply src/main/java/dk/kb/netarchivesuite/solrwayback/parsers/WarcParser.java:267]: Because transfer encoding can contain the ':' character as well.

**Pull request merged**